packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -98,34 +98,39 @@ - (instancetype)initWithQueue:(nonnull SKPaymentQueue *)queue
 }
 
 - (void)startObservingPaymentQueue {
-  self.isObservingTransactions = true;
+  self.observingTransactions = YES;
 
   [self processCachedTransactions];
 }
 
 - (void)stopObservingPaymentQueue {
-  self.isObservingTransactions = false;
+  // Don't remove the transaction observer from the SKPaymentQueue instance as
+  // that will result in transactions being lost during the current application
+  // lifetime. Only setting the "observingTransactions" to "NO" ensures
+  // incoming transactions from the App Store are cached and delivered to the
+  // client as soon as it indicates it is ready to receive transactions by
+  // sending the "startObservingPaymentQueue" message.
+  self.observingTransactions = NO;
 }
 
 - (void)processCachedTransactions {
   NSArray *cachedObjects =
       [self.transactionCache getObjectsForKey:TransactionCacheKeyUpdatedTransactions];
   if (cachedObjects) {
```

Comment:
Minor nit: these checks should probably be `foo.count != 0` rather than a nil check, so that if for whatever reason the cache returned an empty array in the future nothing would get called.

Reply:
That makes a lot of sense. Updated and included an addition test to guard for both (`nil` and empty array) situations.